apps/prairielearn/src/lib/question-render.ts

```diff
@@ -1,49 +1,59 @@
-// @ts-check
-
 import * as async from 'async';
+import type { Response } from 'express';
 import { z } from 'zod';
 
 import * as error from '@prairielearn/error';
 import * as sqldb from '@prairielearn/postgres';
+import { run } from '@prairielearn/run';
 import { generateSignedToken } from '@prairielearn/signed-token';
 
 import { AssessmentScorePanel } from '../components/AssessmentScorePanel.html.js';
 import { QuestionFooter } from '../components/QuestionContainer.html.js';
+import { QuestionContext } from '../components/QuestionContainer.types.js';
 import { QuestionNavSideButton } from '../components/QuestionNavigation.html.js';
 import { QuestionScorePanel } from '../components/QuestionScore.html.js';
 import {
   SubmissionPanel,
   SubmissionBasicSchema,
   SubmissionDetailedSchema,
 } from '../components/SubmissionPanel.html.js';
+import type { SubmissionForRender } from '../components/SubmissionPanel.html.js';
 import { selectVariantsByInstanceQuestion } from '../models/variant.js';
 import * as questionServers from '../question-servers/index.js';
 
 import { config, setLocalsFromConfig } from './config.js';
 import {
+  Assessment,
+  AssessmentInstance,
   AssessmentInstanceSchema,
+  AssessmentQuestion,
   AssessmentQuestionSchema,
   AssessmentSchema,
   AssessmentSetSchema,
+  Course,
   CourseInstanceSchema,
   CourseSchema,
   DateFromISOString,
   GradingJobSchema,
+  GroupConfig,
+  GroupConfigSchema,
   IdSchema,
+  InstanceQuestion,
   InstanceQuestionSchema,
   IssueSchema,
+  Question,
   QuestionSchema,
+  Submission,
   SubmissionSchema,
+  Variant,
   VariantSchema,
 } from './db-types.js';
```

Comment:
I'm thinking that if we care about this enough, we should use https://typescript-eslint.io/rules/consistent-type-imports/ to enforce it, otherwise we should collectively not comment on this in code review. Thoughts?

Reply:
I have become used to having this kind of consistency, and that rule allows for automatic fix, so I'd say let's add this rule. I'm ok with setting this in a separate PR.

Reply:
I opened https://github.com/PrairieLearn/PrairieLearn/pull/10828 to add that lint rule.